core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1309,4 +1309,10 @@ package object config {
     .bytesConf(ByteUnit.BYTE)
     .createOptional
 
+  private[spark] val LEGACY_SESSION_INIT_WITH_DEFAULTS =
+    ConfigBuilder("spark.legacy.sessionInitWithConfigDefaults")
+      .doc("Flag to revert to legacy behavior where a cloned SparkSession receives SparkConf " +
+        "defaults, dropping any overrides in its parent SparkSession.")
+      .booleanConf
+      .createWithDefault(false)
```

Comment:
And, can we move this to `StaticSQLConf.scala` inside `sql/catalyst` module instead of this `core` module?

Reply:
+1

Reply:
Moved and renamed.